botorch/models/approximate_gp.py

```diff
@@ -386,6 +372,85 @@ def __init__(
 
         self.to(train_X)
 
+    def init_inducing_points(
+            self,
+            inputs: Tensor,
+    ) -> Tensor:
+        r"""
+        Reinitialize the inducing point locations in-place with the current kernel
+        applied to `inputs`.
+        The variational distribution and variational strategy caches are reset.
+
+        Args:
+            inputs: input_data (*batch_shape, n, d)
+
+        Returns:
+            (*batch_shape, m, d) selected inducing point locations
+        """
+        var_strat = self.model.variational_strategy
+        clear_cache_hook(var_strat)
+        if hasattr(var_strat, 'base_variational_strategy'):
+            var_strat = var_strat.base_variational_strategy
```

Comment:
Should the caches also be cleared on `base_variational_strategy`? I guess one alternative would be if we made `clear_cache_hook` operate recursively on submodules.

Reply:
I guess I assumed it already operated recursively.